modules/dag-shared/src/main/scala/org/tessellation/dag/snapshot/epoch.scala

```diff
@@ -0,0 +1,36 @@
+package org.tessellation.dag.snapshot
+
+import cats.Order
+import cats.kernel._
+import cats.syntax.contravariant._
+import cats.syntax.semigroup._
+
+import org.tessellation.schema._
+
+import derevo.cats.show
+import derevo.circe.magnolia.{decoder, encoder}
+import derevo.derive
+import eu.timepit.refined.cats._
+import eu.timepit.refined.types.numeric.NonNegLong
+import io.estatico.newtype.macros.newtype
+
+object epoch {
+
+  @derive(encoder, decoder, show)
+  @newtype
+  case class EpochProgress(value: NonNegLong)
+
+  object EpochProgress {
+    val MinValue: EpochProgress = EpochProgress(NonNegLong.MinValue)
+
+    implicit val eqv: Eq[EpochProgress] = Eq[NonNegLong].contramap(_.value)
+    implicit val order: Order[EpochProgress] = Order[NonNegLong].contramap(_.value)
+    implicit val ordering: Ordering[EpochProgress] = order.toOrdering
```

Comment:
These instances can be autogenerated. Please look at the `Height` or `SnapshotOrdinal`.

Reply:
`SnapshotOrdinal`:

```scala
@derive(encoder, decoder, show)
case class SnapshotOrdinal(value: NonNegLong)

object SnapshotOrdinal {
  val MinValue: SnapshotOrdinal = SnapshotOrdinal(NonNegLong.MinValue)

  implicit val order: Order[SnapshotOrdinal] = Order[NonNegLong].contramap(_.value)
  implicit val ordering: Ordering[SnapshotOrdinal] = order.toOrdering
```

From what I remember there was a problem with deriving these values and as you can see `SnapshotOrdinal` also uses `contramap`. But I will investigate the problem. For `Height` it works because it is just `Long` (unless you changed that to `NonNegLong` finally when I was off. I will pull the latest changes).

Reply:
Update: I fixed both `SnapshotOrdinal` and `EpochProgress` so it will be derived automatically.